libraries/chain/block_log.cpp

```diff
@@ -185,6 +185,16 @@ namespace eosio { namespace chain {
          return block;
       }
 
+      template <typename Stream>
+      void read_serialized_block(Stream&& ds, uint64_t block_size, std::vector<char>& buff) {
+         assert(buff.empty());
```

Comment:
The point of taking the vector by reference was to append to the already filled out message header from net_plugin. However, I realize now that the message_header includes the payload size which you don't know until here. 

We could let this method create the exact net_message needed. Call it `vector<char> create_net_message_from_block(block_num_type block_num)` then we could avoid the copy. 

If we don't want to add that knowledge to `block_log.cpp` then I guess we will have to pay for an extra copy of the `vector<char>` or do something more fancy and use two `boost::asio::const_buffer`s, one for the header and one for the payload.

Reply:
I was trying to insert the serialized bytes into the Net message but gave up due to complexities. I like the original returning vector approach: it is cleaner, no precondition assumption, and no knowledge of net plugin.